csharp/src/Ice/Communicator.cs

```diff
@@ -851,45 +845,24 @@ public async ValueTask<IObjectPrx> CreateAdminAsync(ObjectAdapter? adminAdapter,
                 }
 
                 _adminIdentity = adminIdentity;
-                if (adminAdapter == null)
-                {
-                    if (GetProperty("Ice.Admin.Endpoints") != null)
-                    {
-                        adminAdapter = CreateObjectAdapter("Ice.Admin");
-                    }
-                    else
-                    {
-                        throw new InvalidConfigurationException("Ice.Admin.Endpoints is not set");
-                    }
-                }
-                else
+
+                if (adminAdapter != null)
                 {
                     _adminAdapter = adminAdapter;
+                    AddAllAdminFacets();
                 }
-                Debug.Assert(_adminAdapter != null);
-                AddAllAdminFacets();
             }
 
-            if (adminAdapter == null) // the parameter is null which means _adminAdapter needs to be activated
+            if (adminAdapter == null)
             {
-                try
-                {
-                    await _adminAdapter.ActivateAsync().ConfigureAwait(false);
-                }
-                catch
-                {
-                    // We cleanup _adminAdapter, however this error is not recoverable (can't call again GetAdmin()
-                    // after fixing the problem) since all the facets (servants) in the adapter are lost
-                    await _adminAdapter.DisposeAsync().ConfigureAwait(false);
-                    lock (_mutex)
-                    {
-                        _adminAdapter = null;
-                    }
-                    throw;
-                }
+                return await GetAdminAsync(cancel).ConfigureAwait(false) ??
+                    throw new InvalidConfigurationException("missing Ice.Admin.Endpoints");
```

Comment:
It would be better to check for this error condition synchronously before calling GetAdminAsync?

Reply:
Ok, fixed.